### ISSUE TITLE
Daylight saving time not applied for EPG and recordings

### ISSUE THREAD

**Issue by rbuehlma** [xbmc/xbmc#15808]
### Describe the bug
Here is a clear and concise description of what the problem is:

On March 31th 2019 the time gets switched to the daylight saving time. This means that the time will jump from 02:00 to 03:00. This "jump" is not visible in the EPG and all the programmes after 02:00 are displayed as one hour too early. This also affects planned recordings (timers).

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

From my point of view, programmes starting during the daylight saving period should be displayed with the daylight saving offset applied even if its not currently effective.

## Actual Behavior

Programmes and recordings are displayed with one hour offset (too early).

### To Reproduce

1. Set the timezone to e.g. germany
2. Scroll in the EPG to the daylight saving time start (March 31th 02:00)
3. There should be a jump from March 31th 02:00 to March 31th 03:0

### Debuglog

The debuglog can be found here:

I do not expect that a debug log is required here but I verified that the addon pvr.zattoo provides the correct time in UTC for planned timers.

### Screenshots 
Here are some links or screenshots to help explain the problem:

Here the EPG timeline during the daylight saving time switch:
https://imgur.com/a/ZNXmDp6

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ X ] Linux
 - [ ] OSX
 - [ ] Raspberry-Pi
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name:
 - Kodi version: 18.1

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by ksooo** [comment]
This is a bug in Kodi's CDateTime class, which always applies the currently valid UTC offset when converting to local time, not the offset according to the given date.

**Comment by rbuehlma** [comment]
Closing this since it seems to be fixed.

**Comment by ksooo** [comment]
I’m not aware that I or anybody else fixed this.

**Comment by rbuehlma** [comment]
Yes, I just checked it on a different system: Its still an issue -> Reopening it.
